applicationconfig/client/src/test/java/com/azure/applicationconfig/ConfigurationAsyncClientTest.java

```diff
@@ -719,6 +773,82 @@ public void listRevisions() {
+                .assertNext(response -> assertConfigurationEquals(settingUpdate, response))
+                .verifyComplete();
+
+        StepVerifier.create(client.addSetting(setting2))
+                .assertNext(response -> assertConfigurationEquals(setting2, response))
+                .verifyComplete();
+
+        StepVerifier.create(client.updateSetting(setting2Update).delayElement(Duration.ofSeconds(3)))
+                .assertNext(response -> assertConfigurationEquals(setting2Update, response))
+                .verifyComplete();
+
+        StepVerifier.create(client.listSettingRevisions(new SettingSelector().keys(key, key2)).collectList())
+                .assertNext(response -> {
+                    assertTrue(response.stream().anyMatch(result -> configurationsEqual(setting, result)));
+                    assertTrue(response.stream().anyMatch(result -> configurationsEqual(settingUpdate, result)));
+                    assertTrue(response.stream().anyMatch(result -> configurationsEqual(setting2, result)));
+                    assertTrue(response.stream().anyMatch(result -> configurationsEqual(setting2Update, result)));
+                })
+                .verifyComplete();
+    }
+
+    /**
+     * Verifies that we can get all revisions for all settings with the specified labels.
+     */
+    @Test
+    public void listRevisionsWithMultipleLabels() {
+        final String key = sdkContext.randomResourceName(keyPrefix, 16);
+        final String label = sdkContext.randomResourceName(labelPrefix, 16);
+        final String label2 = sdkContext.randomResourceName(labelPrefix, 16);
+        final ConfigurationSetting setting = new ConfigurationSetting().key(key).value(label).label(label);
+        final ConfigurationSetting settingUpdate = new ConfigurationSetting(setting).value(label + "updated");
+        final ConfigurationSetting setting2 = new ConfigurationSetting().key(key).value(label2).label(label2);
+        final ConfigurationSetting setting2Update = new ConfigurationSetting(setting2).value(label2 + "updated");
+
+        StepVerifier.create(client.addSetting(setting))
+                .assertNext(response -> assertConfigurationEquals(setting, response))
+                .verifyComplete();
+
+        StepVerifier.create(client.updateSetting(settingUpdate))
+                .assertNext(response -> assertConfigurationEquals(settingUpdate, response))
+                .verifyComplete();
+
+        StepVerifier.create(client.addSetting(setting2))
+                .assertNext(response -> assertConfigurationEquals(setting2, response))
+                .verifyComplete();
+
+        StepVerifier.create(client.updateSetting(setting2Update))
+                .assertNext(response -> assertConfigurationEquals(setting2Update, response))
+                .verifyComplete();
+
+        StepVerifier.create(client.listSettingRevisions(new SettingSelector().keys(key).labels(label, label2)).collectList())
```

Comment:
A suggestion could be, since you know all the revisions, to create a set of them (ie. `new Set<ConfigurationSetting>() { setting, settingUpdate, seetting2, setting2Update}`) and as each revision is listed to remove them from that `Set<ConfigurationSetting>`. Then, after completion, verify that the set is empty.

It's possible that since you're collecting all the results as a list and seeing if there is anything that matches, the collected list has 5 revisions rather than 4.

Reply:
Done